chord_metadata_service/restapi/authz_middleware.py

```diff
@@ -0,0 +1,37 @@
+from django.conf import settings
+from django.http import HttpResponseServerError
+import requests
+import json
+
+
+class AuthzMiddleware:
+    """
+    Middleware for dataset-level authorization.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if request.path == "/api/phenopackets" and request.method == "GET":
+            request_body = {
+                "user_tokens": [],
+                "path": [request.path],
+                "method": request.method
+            }
+
+            try:
+                response = requests.post(settings.CANDIG_OPA_URL + "/v1/data/permissions", json=request_body)
```

Comment:
if settings.CANDIG_OPA_URL is not set  or None, it returns:
```
response = requests.post(settings.CANDIG_OPA_URL + "/v1/data/permissions", json=request_body)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

you probably need to add a check here to make sure that settings.CANDIG_OPA_URL is not None:

```
 if settings.CANDIG_OPA_URL:
      try:
          response = requests.post(settings.CANDIG_OPA_URL + "/v1/data/permissions", json=request_body)
          response.raise_for_status()
```